File: src/1.SRP/detabase/DatabaseConnectionManager.ts
export default class DatabaseConnectionManager {
    static manager = new DatabaseConnectionManager();

    constructor() {
        
    }

    static getManagerInstance(): DatabaseConnectionManager {
        return DatabaseConnectionManager.manager;
    }

    connect(): void {
        console.log("Connected to DB");
    }

    disconnect(): void {
        console.log("DB Disconnected..")
    }

    getConnectionObject() {
        console.log("Connection Object Sent")
    }
}

File: 1.SRP/src/database/DatabaseConnectionManager.java
package database;

import java.sql.Connection;
import java.sql.SQLException;

public class DatabaseConnectionManager {
    private Connection conn;
    private static DatabaseConnectionManager databaseConnection;
    private DatabaseConnectionManager() {

    }

    public static DatabaseConnectionManager getManagerInstance() {
        if(databaseConnection == null) {
            databaseConnection = new DatabaseConnectionManager();
        }
        return databaseConnection;
    }

    public void connect() throws SQLException {
//        conn = DriverManager.getConnection("/Somewhere/Connect");
        System.out.println("Connect to DB..");
    }

    public Connection getConnectionObject() {
        return conn;
    }

    public void disconnect() throws SQLException {
//        conn.close();
        System.out.println("Disconnect from db");
    }
}
